src/Protocol/DocumentUri.cs

```diff
@@ -28,7 +28,7 @@ namespace OmniSharp.Extensions.LanguageServer.Protocol
     /// </summary>
     /// <remarks>This exists because of some non-standard serialization in vscode around uris and .NET's behavior when deserializing those uris</remarks>
     [JsonConverter(typeof(DocumentUriConverter))]
-    public partial class DocumentUri : IEquatable<DocumentUri?>
+    public partial record DocumentUri
```

Comment:
Man it's pretty dope that records just make things equatable

Reply:
and actually... this might need to revert this..  because of case sensitive file systems.